index.bs

```diff
@@ -837,11 +837,26 @@ When an {{XRReferenceSpace}} is requested, the user agent MUST <dfn>create a ref
 
   1. Let |session| be the {{XRSession}} object that requested creation of a reference space.
   1. Let |type| be set to the {{XRReferenceSpaceType}} passed to {{requestReferenceSpace()}}.
-  1. If |type| is {{bounded-floor}}, let |referenceSpace| be a new {{XRBoundedReferenceSpace}}.
-  1. Else let |referenceSpace| be a new {{XRReferenceSpace}}.
-  1. Initialize |referenceSpace|'s [=XRReferenceSpace/type=] to be |type|.
-  1. Initialize [=XRSpace/session=] to be |session|.
-  1. Return |referenceSpace|.
+  1. If the [=reference space is supported=] for |type| and |session|, run the following steps:
+    1. If |type| is {{bounded-floor}}, let |referenceSpace| be a new {{XRBoundedReferenceSpace}}.
+    1. Else let |referenceSpace| be a new {{XRReferenceSpace}}.
+    1. Initialize |referenceSpace|'s [=XRReferenceSpace/type=] to be |type|.
+    1. Initialize |referenceSpace|'s [=XRSpace/session=] to be |session|.
+    1. Return |referenceSpace|
+  1. Else, return <code>null</code>.
+
+</div>
+
+<div class="algorithm" data-algorithm="reference-space-supported">
+To check if a <dfn>reference space is supported</dfn> for a given reference space type |type| and {{XRSession}} |session|, run the following steps:
+
+
+  1. If |type| is {{inline}}, return <code>true</code>.
+  1. If |type| is {{local}} or {{local-floor}}, and |session| is an [=immersive session=], return <code>true</code>.
+  1. If |type| is {{local}} or {{local-floor}}, and the [=/XR device=] supports reporting orientation data, return <code>true</code>.
```

Comment:
To confirm, apps should be able to get a `local` reference space even before the session becomes immersive?

Reply:
Sessions do not "become" immersive, they are a single mode for their lifetime. Inline sessions should be able to create `local` reference spaces, assuming the UA has properly handled the privacy implications (ongoing discussion).